doc/DataPlaneCodeGeneration/EvolutionArchBoard.md

```diff
@@ -0,0 +1,630 @@
+```C#
+public virtual Pageable<BinaryData> GetMetricDimension(string metricId, RequestContent content, int? skip = null, int? maxpagesize = null, string optional = null, RequestContext context = null);
+```
+Breaking scenario in v<sub>i+1</sub>
+```C#
+Pageable<BinaryData> = GetMetricDimension(metricId, RequestContent.Create(content), skip, maxpagesize, new RequestContext());
+```
+Solution in v<sub>i+1</sub>
+```C#
+public virtual Pageable<BinaryData> GetMetricDimension(string metricId, RequestContent content, int? skip, int? maxpagesize, RequestContext context);
+public virtual Pageable<BinaryData> GetMetricDimension(string metricId, RequestContent content, int? skip = null, int? maxpagesize = null, string optional = null, RequestContext context = null);
+```
+### Case 2: A method changes a required parameter to optional
+Service changes a required parameter to optional.
+#### Case 2.1: Required input is changed to an optional input in a method with no other parameters ([APIView](https://apiview.dev/Assemblies/Review/c8b0b44916bb4517ac0c9fa49d9a7f37?diffRevisionId=f6c400a77e8d407e994b477c5bdb6efa&doc=False&diffOnly=False&revisionId=735fa276d20e458384447336226467b7) Lines 22-23)
+Generated code in v<sub>i</sub>
+```C#
+public virtual Response DeleteGroup(string groupId, RequestContext context = null);
+```
+Generated code in v<sub>i+1</sub> after changing `groupId` to optional
+```C#
+public virtual Response DeleteGroup(string groupId = null, RequestContext context = null);
+```
+Breaking scenario in v<sub>i+1</sub>
+
+No. Safe to go.
+#### Case 2.2: Required input is changed to an optional input when it is in the last position in the path parameter ([APIView](https://apiview.dev/Assemblies/Review/c8b0b44916bb4517ac0c9fa49d9a7f37?diffRevisionId=f6c400a77e8d407e994b477c5bdb6efa&doc=False&diffOnly=False&revisionId=735fa276d20e458384447336226467b7) Lines19-20)
+Generated code in v<sub>i</sub>
+```C#
+public virtual Response DeleteDeployment(string groupId, string deploymentId, RequestContext context = null);
+```
+Generated code in v<sub>i+1</sub> after changing `deploymentId` to optional
+```C#
+public virtual Response DeleteDeployment(string groupId, string deploymentId = null, RequestContext context = null);
+```
+Breaking scenario in v<sub>i+1</sub>
+
+No. Safe to go.
+#### Case 2.3: Required input is changed to an optional input when it is in a non-last position
+Generated code in v<sub>i</sub>
+```C#
+public virtual Response DeleteDeployment(string groupId, string deploymentId, RequestContext context = null);
+```
+Generated code in v<sub>i+1</sub> after changing `groupId` to optional
+```C#
+public virtual Response DeleteDeployment(string deploymentId, string groupId = null, RequestContext context = null);
+```
+Breaking scenario in v<sub>i+1</sub>
+```C#
+Response response = DeleteDeployment(groupId, deploymentId);
+```
+Solution in v<sub>i+1</sub>
+
+Solve it manually.
```

Comment:
As i know, there is a feedback in the review that we should report error instead of ask people to solve it manually, do I understand correctly?

Reply:
It is correct we should not solve it manually. But the solution still needs discussion. Report error is a possible solution. 
Do we need to update this page with the feedback? There are many things still need to discuss.